js/core/game.js

```diff
@@ -520,25 +520,11 @@ define(["dao", "globals", "ui", "core/freeAgents", "core/finances", "core/gameSi
                     p.ovr = rating.ovr;
 
                     // These use the same formulas as the skill definitions in player.skills!
-                    p.compositeRating.pace = makeComposite(rating, ['spd', 'jmp', 'dnk', 'tp', 'stl', 'drb', 'pss']);
-                    p.compositeRating.usage = Math.pow(makeComposite(rating, ['ins', 'dnk', 'fg', 'tp', 'spd', 'drb'], [1.5, 1, 1, 1, 0.15, 0.15]), 1.9);
-                    p.compositeRating.dribbling = makeComposite(rating, ['drb', 'spd']);
-                    p.compositeRating.passing = makeComposite(rating, ['drb', 'pss'], [0.4, 1]);
-                    p.compositeRating.turnovers = makeComposite(rating, ['drb', 'pss', 'spd', 'hgt', 'ins'], [1, 1, -1, 1, 1]);  // This should not influence whether a turnover occurs, it should just be used to assign players
-                    p.compositeRating.shootingAtRim = makeComposite(rating, ['hgt', 'spd', 'jmp', 'dnk'], [1, 0.2, 0.6, 0.4]);  // Dunk or layup, fast break or half court
-                    p.compositeRating.shootingLowPost = makeComposite(rating, ['hgt', 'stre', 'spd', 'ins'], [1, 0.6, 0.2, 1]);  // Post scoring
-                    p.compositeRating.shootingMidRange = makeComposite(rating, ['hgt', 'fg'], [0.2, 1]);  // Two point jump shot
-                    p.compositeRating.shootingThreePointer = makeComposite(rating, ['hgt', 'tp'], [0.2, 1]);  // Three point jump shot
-                    p.compositeRating.shootingFT = makeComposite(rating, ['ft']);  // Free throw
-                    p.compositeRating.rebounding = makeComposite(rating, ['hgt', 'stre', 'jmp', 'reb'], [1.5, 0.1, 0.1, 0.7]);
-                    p.compositeRating.stealing = makeComposite(rating, ['constant', 'spd', 'stl'], [1, 1, 1]);
-                    p.compositeRating.blocking = makeComposite(rating, ['hgt', 'jmp', 'blk'], [1.5, 0.5, 0.5]);
-                    p.compositeRating.fouling = makeComposite(rating, ['constant', 'hgt', 'blk', 'spd'], [1.5, 1, 1, -1]);
-                    p.compositeRating.defense = makeComposite(rating, ['hgt', 'stre', 'spd', 'jmp', 'blk', 'stl'], [1, 1, 1, 0.5, 1, 1]);
-                    p.compositeRating.defenseInterior = makeComposite(rating, ['hgt', 'stre', 'spd', 'jmp', 'blk'], [2, 1, 0.5, 0.5, 1]);
-                    p.compositeRating.defensePerimeter = makeComposite(rating, ['hgt', 'stre', 'spd', 'jmp', 'stl'], [1, 1, 2, 0.5, 1]);
-                    p.compositeRating.endurance = makeComposite(rating, ['constant', 'endu', 'hgt'], [1, 1, -0.1]);
-                    p.compositeRating.athleticism = makeComposite(rating, ['stre', 'spd', 'jmp', 'hgt'], [1, 1, 1, 0.5]); // Currently only used for synergy calculation
+                    for (k in g.compositeWeights) {
+                        if ({}.hasOwnProperty.call(g.compositeWeights, k)) {
```

Comment:
A much clearer way to write the same thing:

    if (g.compositeWeights.hasOwnProperty.(k)) {